core/src/main/resources/error/error-classes.json

```diff
@@ -542,6 +542,12 @@
     ],
     "sqlState" : "22003"
   },
+  "ARRAY_INSERT_BY_INDEX_ZERO" : {
+    "message" : [
+      "The index 0 is invalid. An index shall be either < 0 or > 0 (the first element has index 1)."
+    ],
+    "sqlState" : "22003"
```

Comment:
I just reuse the `sqlState` of  `ELEMENT_AT_BY_INDEX_ZERO`, is it fine? @itholic

Reply:
Yes, it sounds good to me.

But on my second thought, maybe can we consolidate `ELEMENT_AT_BY_INDEX_ZERO` and `ARRAY_INSERT_BY_INDEX_ZERO` into single error class if they have exactly the same error message??

For example:
```
  "ZERO_INDEX_ERROR" : {
    "message" : [
      "The index 0 is invalid. An index shall be either < 0 or > 0 (the first element has index 1)."
    ],
    "sqlState" : "22003"
  },
```

I'm not 100% sure if `ZERO_INDEX_ERROR` is proper to cover the both cases. WDYT @srielau @MaxGekk ?

Reply:
how about `INVALID_INDEX_OF_ZERO`?

Reply:
Sounds good. We need to start hardening the error classes, though. They are documented API.